testing/it_sidecar/client/sidecar_client.go

```diff
@@ -26,8 +26,14 @@ type K8STestSetup struct {
 	in  io.WriteCloser
 	out io.ReadCloser
 	er  io.ReadCloser
+
+	// ReadyCallbacks for custom setup after services are ready and before pre-test
+	ReadyCallbacks []Callback
```

Comment:
I'd make it just one function `ReadyCallback Callback`. it is easy to define `func` you need to notify multiple observers.